core/src/main/java/nl/nn/adapterframework/pipes/IteratingPipe.java

```diff
@@ -414,18 +414,34 @@ public void waitForResults() throws SenderException, IOException {
 				try {
 					guard.waitForAllResources();
 					int count = 0;
+					List<Throwable> exceptions = new ArrayList<>();
 					for (ParallelSenderExecutor pse : executorList) {
 						count++;
-						String itemResult;
+						String itemResult="";
 						if (pse.getThrowable() == null) {
 							SenderResult senderResult = pse.getReply();
 							if (senderResult.isSuccess()) {
 								itemResult = senderResult.getResult().asString();
-							} else {
-								itemResult = "<exception>"+ XmlEncodingUtils.encodeChars(senderResult.getResult().asString())+"</exception>";
+							} 
+							else {
+								if(isIgnoreExceptions()){
+									itemResult = "<exception>"+ XmlEncodingUtils.encodeChars(senderResult.getResult().asString())+"</exception>";
+								} else {
+									SenderException se = new SenderException(senderResult.getResult().asString());
+									exceptions.add(se);
```

Comment:
Although it's quite minor, this could be on a single line.
```suggestion
									exceptions.add(new SenderException(senderResult.getResult().asString()));
```